docs/customresourcestate-metrics.md

```diff
@@ -272,19 +267,53 @@ spec:
+          help: "VPA container recommendations for memory."
           each:
             type: Gauge
             gauge:
-              path: [metadata, annotations]
-# This will output the following metric:
-# HELP kube_customresource_autoscaling_annotations Kubernetes annotations converted to Prometheus labels.
-# TYPE kube_customresource_autoscaling_annotations gauge
-# kube_customresource_autoscaling_annotations{customresource_group="autoscaling.k8s.io", customresource_kind="VerticalPodAutoscaler", customresource_version="v1", namespace="default",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",verticalpodautoscaler="hamster-vpa"} 123
+              path: [status, recommendation, containerRecommendations]
+              valueFrom: [target, memory]
+              labelsFromPath:
+                container: [containerName]
+          commonLabels:
+            resource: "memory"
+            unit: "byte"
+        - name: "vpa_containerrecommendations_target"
+          help: "VPA container recommendations for cpu."
+          each:
+            type: Gauge
+            gauge:
+              path: [status, recommendation, containerRecommendations]
+              valueFrom: [target, cpu]
+              labelsFromPath:
+                container: [containerName]
+          commonLabels:
+            resource: "cpu"
+            unit: "core"
+```
+
+The above configuration was tested on [this](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/examples/hamster.yaml) VPA configuration, outputting the following metrics:
+
+```
+# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for memory.
+kube_customresource_vpa_containerrecommendations_target{container="hamster",customresource_group="autoscaling.k8s.io",customresource_kind="VerticalPodAutoscaler",customresource_version="v1",namespace="default",resource="memory",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",unit="byte",verticalpodautoscaler="hamster-vpa"} 1.048576e+08
+
+# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for cpu.
+kube_customresource_vpa_containerrecommendations_target{container="hamster",customresource_group="autoscaling.k8s.io",customresource_kind="VerticalPodAutoscaler",customresource_version="v1",namespace="default",resource="cpu",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",unit="core",verticalpodautoscaler="hamster-vpa"} 0.1
 ```
 
-The above configuration was tested on [this](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/examples/hamster.yaml) VPA configuration, with an added annotation (`foo: 123`).
+To ensure that the VPA Metrics will be captured seamlessly, make sure that the Cluster Role / Role assigned to the Kube State Metrics has permissions to `list` & `watch` VPA resources e.g.:
+
+```yaml
```

Comment:
seems like the Cluster Role/ Role assigned to KSM requires additional permissions to CRDs

```
- verbs:
     - list
     - watch
   apiGroups:
     - apiextensions.k8s.io
   resources:
     - customresourcedefinitions
```

otherwise `verticalpodautoscalers` still won't be available 

```
E1030 12:35:19.531362       1 reflector.go:148] pkg/mod/k8s.io/client-go@v0.27.5/tools/cache/reflector.go:231: Failed to watch apiextensions.k8s.io/v1, Resource=customresourcedefinitions: failed to list apiextensions.k8s.io/v1, Resource=customresourcedefinitions: customresourcedefinitions.apiextensions.k8s.io is forbidden: User "system:serviceaccount:monitoring:kube-prometheus-stack-kube-state-metrics" cannot list resource "customresourcedefinitions" in API group "apiextensions.k8s.io" at the cluster scope
```

Reply:
@k0nstantinv why do you need CRDs there? .. It's only regarding the VPA stuff and there I am not sure if you need anything other than the `verticalpodautoscalers` resource in the `autoscaling.k8s.io` group ..

Reply:
Now I get where my confusion comes from. I am using the Helm Chart directly to install KSM and hence, I just needed to add the VPA stuff .. I re-committed with the CRD RBAC permissions also added. Please have another look.

Reply:
Thanks! looks ok